qiskit/chemistry/algorithms/pes_samplers/potentials/morse_potential.py

```diff
@@ -0,0 +1,247 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+This module implements a 1D Morse potential.
+"""
+
+from typing import List
+import numpy as np
+from scipy.optimize import curve_fit
+
+from qiskit.chemistry.algorithms.pes_samplers.potentials.potential_base import PotentialBase
+from qiskit.chemistry.drivers import Molecule
+import qiskit.chemistry.constants as const
+
+
+class MorsePotential(PotentialBase):
+    """
+    Implements a 1D Morse potential.
+    Input units are Angstroms (distance between the two atoms),
+        and output units are Hartrees (molecular energy).
+    """
+    # Works in Angstroms and Hartrees
+
+    def __init__(self, molecule):
+        """
+        Initializes the potential to the zero-function.
+        fit() should be used afterwards to fit the potential to
+            computed molecular energies.
+
+        Args:
+            molecule: the underlying molecule.
+
+        Raises:
+            ValueError: Only implemented for diatomic molecules
+        """
+        # Initialize with zero-potential.
+        # Later - fit energy values (fit)
+        self.d_e = 0.0
+        self.m_shift = 0.0
+        self.alpha = 0.0
+        self.r_0 = 0.0
+        self._m_a = molecule.masses[0]
```

Comment:
This code seems problematic. masses in Molecule is Optional so the masses getter can return None. Elsewhere in the code there is at least a length check (albeit that has a similar issue) on update_molecule. Arguable should so a None check here and then check len and raise an error. Same in update molecule

Reply:
Having looked more at the update_molecule and seen it defined in a parent class of this, which also has an init (ie VibronicStructureBase) it seems that init would not be called. Vib... Base is a  parent of PotentialBase which this class implements

Reply:
I fixed the check for the masses.